src/wallet/reconstruct_wallet_flow.c

```diff
@@ -333,20 +333,33 @@ bool reconstruct_seed_flow(const uint8_t *wallet_id, uint8_t *seed_out) {
 
   clear_wallet_data();
 
-  if (COMPLETED ==
-      reconstruct_wallet_secret_flow(wallet_id, secret, PASSPHRASE_INPUT)) {
+  // Select wallet based on wallet_id
+  if (!get_wallet_data_by_id(wallet_id, &wallet, reject_cb)) {
+    return false;
+  }
+
+  reconstruct_state_e flow =
+      reconstruct_secret(wallet_id, secret, PASSPHRASE_INPUT);
+
+  if (COMPLETED == flow) {
     get_seed_from_secret(secret, seed_out);
     result = true;
+  } else if (reject_cb && EARLY_EXIT == flow) {
+    // Inform the host of any rejection
+    reject_cb(ERROR_COMMON_ERROR_USER_REJECTION_TAG,
+              ERROR_USER_REJECTION_CONFIRMATION);
+  } else if (reject_cb && COMPLETED_WITH_ERRORS == flow) {
+    // Inform the host of any card error
+    reject_cb(ERROR_COMMON_ERROR_CARD_ERROR_TAG, 0);
```

Comment:
Let's use the enum `WALLET_NOT_FOUND_UNKNOWN` or `WALLET_PARTIAL_STATE_UNKNOWN` for reference.

Reply:
Ok, updated